### ISSUE TITLE
Vscode plugin used in remote connection

### ISSUE THREAD

**Issue by zhenyuwei99** [molstar/VSCoding-Sequence#30]
Hi,

I am new to using the protein viewer plugin in the vscode. I think it would be cool if I could use it on remote-connected Windows so that I don't need to download it to my local host. However, when I start the protein viewer and use the open files, the file system still shows my local file systems. Are there any methods to fix this?

I also tried to use the command "Launch Protein Viewer from File(s)" and threw the error:
<Cannot read properties of undefined (reading 'map')>

Thanks

**Comment by a-r-j** [comment]
Hi @zhenyuwei99 

I agree. I'll take a look but I'm not sure how easy it will be to give the viewer access to the remote file system.

For the command: that command should only be used when right-clicking on a file in the file tree. The command to use from the command panel is: `Start Protein Viewer`.

**Comment by zhenyuwei99** [comment]
Great!  Thank you for your replay.

Right-clicking the file and starting the protein viewer can make me access the PDB file in the remote window. It looks very nice. But it still cannot open other files like load psf files after opening the pdb file.

**Comment by a-r-j** [comment]
> Right-clicking the file and starting the protein viewer can make me access the PDB file in the remote window. It looks very nice. 

Great, I'm glad :D


> But it still cannot open other files like load psf files after opening the pdb file.

Ah, yes. I think I need to update the file extensions that trigger the extension. One workaround though is if you have the desired files on your local machine you should be able to load/open them from within the viewer itself.

**Comment by zhenyuwei99** [comment]
> One workaround though is if you have the desired files on your local machine you should be able to load/open them from within the viewer itself.

Ofc, I can do that on my local machine. But I think it will be very cool to check the correctness of the simulation on the remote device directly without downloading the file, which may be wrong on the local machine. I think that will make this extension super convenient, which can not be done for any other software. It likes to provide a GUI for the remote devices in VSCode. I think updating the file extension might be a partial solution as, most of the time, we hope to read the PSF and DCD files simultaneously as the DCD file does not have the topology information. From the right-click manner, I can only open one file at one time, and it is impossible to load other files after launching the protein viewer as the file system is still looking at the device on my local machine. I think if it is possible to let us load the file on the remote device, the extension will be perfect!

**Comment by a-r-j** [comment]
> Ofc, I can do that on my local machine. But I think it will be very cool to check the correctness of the simulation on the remote device directly without downloading the file, which may be wrong on the local machine. I think that will make this extension super convenient, which can not be done for any other software. 

I agree this would be a nice feature. The issue is I'm not exactly sure how to go about doing it as the extension is running locally. Might I suggest trying something like `sshfs` to mount the local remote filesystem? I'm not 100% sure if it will work but I think it's the easiest thing to try first.

> From the right-click manner, I can only open one file at one time, and it is impossible to load other files after launching the protein viewer as the file system is still looking at the device on my local machine

Is this due to the file extsnsion? You should, for example, be able to select multiple PDB files (Shift/Cmd + click to select) and open them all at once.

**Comment by zhenyuwei99** [comment]
> Might I suggest trying something like sshfs to mount the local remote filesystem?

Maybe? I am not so sure. When I use the remote SSH in vscode to connect the remote host, the file system will be like the path shown in the command plate. Since the 

> You should, for example, be able to select multiple PDB files (Shift/Cmd + click to select) and open them all at once.


If I can do this,  updating the file extension can solve my questions since I can choose the .dcd file and .psf file to launch the Protein Viewer. But it just cannot load other information after launching the protein viewer. Anyway, this will be one of the solutions.

**Comment by a-r-j** [comment]
Ok, I'll try a few things and see what I come up with. Could you share some example files I can use for prototyping?

**Comment by zhenyuwei99** [comment]
Ofc, a Google Drive [link](https://drive.google.com/drive/folders/1Wv_shuDTGBWpBa00xZ_V571PrHWxfS22?usp=sharing) with a .psf file and a .dcd file is here.

By the way, I noticed that in Mol*, many file format is [supported](https://molstar.org/docs/plugin/file-formats/). Could all the file formats supported by Mol* be supported in this plugin?
